### ISSUE TITLE
Avoid installing the `init.d` service if it is not necessary

### ISSUE THREAD

**Issue by rauldpm** [wazuh/wazuh-packages#1918]
# Description

Due to what was reported during the development of the issue https://github.com/wazuh/wazuh-packages/issues/1880, it is necessary to decide on a solution for the `init.d` problem on recent systems such as `Red Hat 9`, `CentOS 9 Stream`, and `Fedora 36` in a more generic way that not only covers the recommended systems, but also those that do not appear on said list and that may, in the future, be part of it.

Due to the deprecation of `init.d` on these systems, there are certain features that are not available when using the `service` binary, specifically, these new systems do not contain a script file that defines certain actions such as `start`, `stop`, `status` and `daemon`. This is due to two facts:

1. The directory `/etc/init.d` is no longer a symlink of `/etc/rc.d/init.d` nor does it exist as a directory.
2. The `/etc/rc.d/init.d/functions` script is no longer available.

As explained in the mentioned issue, the `service` binary establishes a conditional structure in which, if the service exists in `/etc/init.d` (redirection from `/etc/rc.d/init.d`), said `functions` script is used, since there is no such symlink, the service, although it exists in `/etc/rc.d/init.d`, is not used. The problem is that certain packages could restore said symlink because they have not adapted to this new structure, such as `OpenSearch`, which continues to install its service in the `/etc/init.d` directory directly.

For all this, it is necessary to find the most appropriate way to not install the `init.d` service on systems where it is being deprecated. One of the proposed solutions would consist of not installing the service if the `function` script does not exist.

# Tasks

- [x] Investigate options to avoid installing the `wazuh-manager`, `wazuh-agent`, `wazuh-dashboard`, and `wazuh-indexer` services on systems that do not use `init.d`.
- [x] Apply the most viable option.
- [x] Validate changes in the different RPM systems.
  
 
# Validation

- [x] The fix works on the following rpm systems:
  - CentOS 9 Stream
    - [x] Wazuh server
    - [x] Wazuh agent
    - [x] Wazuh dashboard
    - [x] Wazuh indexer
  - Alma Linux 9
    - [x] Wazuh server
    - [x] Wazuh agent
    - [x] Wazuh dashboard
    - [x] Wazuh indexer
  - Fedora 36
    - [x] Wazuh server
    - [x] Wazuh agent
    - [x] Wazuh dashboard
    - [x] Wazuh indexer
  - Red Hat 9
    - [x] Wazuh server
    - [x] Wazuh agent
    - [x] Wazuh dashboard
    - [x] Wazuh indexer
  - CentOS 6
    - [x] Wazuh server
    - [x] Wazuh agent
    - [ ] Wazuh dashboard
    - [ ] Wazuh indexer
  - CentOS 7
    - [x] Wazuh server
    - [x] Wazuh agent
    - [x] Wazuh dashboard
    - [x] Wazuh indexer
  - Red Hat 7
    - [x] Wazuh server
    - [x] Wazuh agent
    - [x] Wazuh dashboard
    - [x] Wazuh indexer
  - Fedora 34
    - [x] Wazuh server
    - [x] Wazuh agent
    - [x] Wazuh dashboard
    - [x] Wazuh indexer
- [x] The installation is successful on the above systems.
- [x] The upgrade is successful on the above systems.

**Comment by verdx** [comment]
# Update - investigation
- After testing the different packages on CentOS 9 Stream the error only reproduces on packages `wazuh-agent` and `wazuh-manager`, at least with version `4.5`.
- There doesn't seem to be a way to clearly discern if a system is running or not some version of `init.d` except the clear one, to check if the problematic file `/urs/lib/systemd/systemd-sysv-install` is present. Packages for `wazuh-agent` and `wazuh-manager` are being created where the solution given in Issues https://github.com/wazuh/wazuh-packages/issues/1626 https://github.com/wazuh/wazuh-packages/pull/1675 of removing `/etc/rc.d/init.d/wazuh-*` is done before. The installer checks if file `/urs/lib/systemd/systemd-sysv-install` is present and if it is not, `/etc/rc.d/init.d/wazuh-*` is not installed in the first place.

**Comment by verdx** [comment]
# Update - investigation

## Test 1
To set the current status of the `systemd` and `sysv` services, a test has been done and run on CentOS 9 Stream, where no patch has been added, not as with [RHEL 9](https://github.com/wazuh/wazuh-packages/pull/1673/files). 
### Test 1 Code
```
WAZUH_MANAGER="10.0.0.2"
for COMPONENT in "wazuh-agent" "wazuh-manager" "wazuh-indexer" "wazuh-dashboard"
do
    yum install -y ./$COMPONENT*
    systemctl daemon-reload
    echo "Enable $COMPONENT"
    systemctl enable $COMPONENT
    echo "Disable $COMPONENT"
    systemctl disable $COMPONENT
    echo "Start $COMPONENT"
    systemctl start $COMPONENT
    echo "Restart $COMPONENT"
    systemctl restart $COMPONENT
    echo "Stop $COMPONENT"
    systemctl stop $COMPONENT
    echo "Service start $COMPONENT"
    service $COMPONENT start
    echo "Service stop $COMPONENT"
    service $COMPONENT stop
    echo "Service status $COMPONENT"
    service $COMPONENT status
    yum remove -y $COMPONENT
done
```

### Test 1 Results
<details>
  <summary>Wazuh agent</summary>

```
Enable wazuh-agent
Synchronizing state of wazuh-agent.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install enable wazuh-agent
Failed to execute /usr/lib/systemd/systemd-sysv-install: No such file or directory
Disable wazuh-agent
Synchronizing state of wazuh-agent.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install disable wazuh-agent
Failed to execute /usr/lib/systemd/systemd-sysv-install: No such file or directory
Start wazuh-agent
Job for wazuh-agent.service failed because the control process exited with error code.
See "systemctl status wazuh-agent.service" and "journalctl -xeu wazuh-agent.service" for details.
Restart wazuh-agent
Job for wazuh-agent.service failed because the control process exited with error code.
See "systemctl status wazuh-agent.service" and "journalctl -xeu wazuh-agent.service" for details.
Stop wazuh-agent
Service start wazuh-agent
Redirecting to /bin/systemctl start wazuh-agent.service
Job for wazuh-agent.service failed because the control process exited with error code.
See "systemctl status wazuh-agent.service" and "journalctl -xeu wazuh-agent.service" for details.
Service stop wazuh-agent
Redirecting to /bin/systemctl stop wazuh-agent.service
Service status wazuh-agent
Redirecting to /bin/systemctl status wazuh-agent.service
× wazuh-agent.service - Wazuh agent
     Loaded: loaded (/usr/lib/systemd/system/wazuh-agent.service; disabled; preset: disabled)
     Active: failed (Result: exit-code) since Fri 2023-02-03 10:59:12 UTC; 50ms ago
    Process: 10939 ExecStart=/usr/bin/env /var/ossec/bin/wazuh-control start (code=exited, status=1/FAILURE)
        CPU: 22ms

Feb 03 10:59:12 centos9stream systemd[1]: Starting Wazuh agent...
Feb 03 10:59:12 centos9stream env[10947]: 2023/02/03 10:59:12 wazuh-agentd: ERROR: (4112): Invalid server address found: 'MANAGER_IP'
Feb 03 10:59:12 centos9stream env[10947]: 2023/02/03 10:59:12 wazuh-agentd: CRITICAL: (1215): No client configured. Exiting.
Feb 03 10:59:12 centos9stream env[10939]: wazuh-agentd: Configuration error. Exiting
Feb 03 10:59:12 centos9stream systemd[1]: wazuh-agent.service: Control process exited, code=exited, status=1/FAILURE
Feb 03 10:59:12 centos9stream systemd[1]: wazuh-agent.service: Failed with result 'exit-code'.
Feb 03 10:59:12 centos9stream systemd[1]: Failed to start Wazuh agent.
```
</details>

<details>
  <summary>Wazuh manager</summary>

```
Enable wazuh-manager
Synchronizing state of wazuh-manager.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install enable wazuh-manager
Failed to execute /usr/lib/systemd/systemd-sysv-install: No such file or directory
Disable wazuh-manager
Synchronizing state of wazuh-manager.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install disable wazuh-manager
Failed to execute /usr/lib/systemd/systemd-sysv-install: No such file or directory
Start wazuh-manager
Restart wazuh-manager
Stop wazuh-manager
Service start wazuh-manager
Redirecting to /bin/systemctl start wazuh-manager.service
Service stop wazuh-manager
Redirecting to /bin/systemctl stop wazuh-manager.service
Service status wazuh-manager
Redirecting to /bin/systemctl status wazuh-manager.service
○ wazuh-manager.service - Wazuh manager
     Loaded: loaded (/usr/lib/systemd/system/wazuh-manager.service; disabled; preset: disabled)
     Active: inactive (dead)

Feb 03 11:01:38 centos9stream env[13340]: Killing wazuh-authd...
Feb 03 11:01:39 centos9stream env[13340]: wazuh-agentlessd not running...
Feb 03 11:01:39 centos9stream env[13340]: wazuh-integratord not running...
Feb 03 11:01:39 centos9stream env[13340]: wazuh-dbd not running...
Feb 03 11:01:39 centos9stream env[13340]: wazuh-csyslogd not running...
Feb 03 11:01:39 centos9stream env[13340]: Killing wazuh-apid...
Feb 03 11:01:39 centos9stream env[13340]: Wazuh v4.5.0 Stopped
Feb 03 11:01:39 centos9stream systemd[1]: wazuh-manager.service: Deactivated successfully.
Feb 03 11:01:39 centos9stream systemd[1]: Stopped Wazuh manager.
Feb 03 11:01:39 centos9stream systemd[1]: wazuh-manager.service: Consumed 18.409s CPU time.
```
</details>

<details>
  <summary>Wazuh indexer</summary>

```
Enable wazuh-indexer
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-indexer.service → /usr/lib/systemd/system/wazuh-indexer.service.
Disable wazuh-indexer
Removed "/etc/systemd/system/multi-user.target.wants/wazuh-indexer.service".
Start wazuh-indexer
Job for wazuh-indexer.service failed because the control process exited with error code.
See "systemctl status wazuh-indexer.service" and "journalctl -xeu wazuh-indexer.service" for details.
Restart wazuh-indexer
Job for wazuh-indexer.service failed because the control process exited with error code.
See "systemctl status wazuh-indexer.service" and "journalctl -xeu wazuh-indexer.service" for details.
Stop wazuh-indexer
Service start wazuh-indexer
Starting wazuh-indexer: /bin/systemctl
/etc/init.d/wazuh-indexer: line 89: daemon: command not found

Service stop wazuh-indexer
Service status wazuh-indexer
/etc/init.d/wazuh-indexer: line 124: status: command not found
```
</details>

<details>
  <summary>Wazuh dashboard</summary>

```
Enable wazuh-dashboard
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-dashboard.service → /etc/systemd/system/wazuh-dashboard.service.
Disable wazuh-dashboard
Removed "/etc/systemd/system/multi-user.target.wants/wazuh-dashboard.service".
Start wazuh-dashboard
Restart wazuh-dashboard
Stop wazuh-dashboard
Service start wazuh-dashboard
Redirecting to /bin/systemctl start wazuh-dashboard.service
Service stop wazuh-dashboard
Redirecting to /bin/systemctl stop wazuh-dashboard.service
Service status wazuh-dashboard
Redirecting to /bin/systemctl status wazuh-dashboard.service
○ wazuh-dashboard.service - wazuh-dashboard
     Loaded: loaded (/etc/systemd/system/wazuh-dashboard.service; disabled; preset: disabled)
     Active: inactive (dead)

Feb 03 11:06:45 centos9stream systemd[1]: wazuh-dashboard.service: Deactivated successfully.
Feb 03 11:06:45 centos9stream systemd[1]: Stopped wazuh-dashboard.
Feb 03 11:06:45 centos9stream systemd[1]: Started wazuh-dashboard.
Feb 03 11:06:45 centos9stream systemd[1]: Stopping wazuh-dashboard...
Feb 03 11:06:45 centos9stream systemd[1]: wazuh-dashboard.service: Deactivated successfully.
Feb 03 11:06:45 centos9stream systemd[1]: Stopped wazuh-dashboard.
Feb 03 11:06:45 centos9stream systemd[1]: Started wazuh-dashboard.
Feb 03 11:06:45 centos9stream systemd[1]: Stopping wazuh-dashboard...
Feb 03 11:06:45 centos9stream systemd[1]: wazuh-dashboard.service: Deactivated successfully.
Feb 03 11:06:45 centos9stream systemd[1]: Stopped wazuh-dashboard.
```
</details>

### Summary of errors in test 1
There are two different errors happening:
- When trying to enable services `wazuh-agent` and `wazuh-manager`, the following error is encountered. This error is fixed as a patch in Issue https://github.com/wazuh/wazuh-packages/issues/1626 by removing file `/etc/rc.d/init.d/wazuh-*`.
```
Enable wazuh-agent
Synchronizing state of wazuh-agent.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install enable wazuh-agent
Failed to execute /usr/lib/systemd/systemd-sysv-install: No such file or directory
Disable wazuh-agent
Synchronizing state of wazuh-agent.service with SysV service script with /usr/lib/systemd/systemd-sysv-install.
Executing: /usr/lib/systemd/systemd-sysv-install disable wazuh-agent
Failed to execute /usr/lib/systemd/systemd-sysv-install: No such file or directory
``` 
- The second error appears when trying to start or get the status of the Wazuh indexer, where the problem explained in the Issue description appears, and some functions cannot be found.
```
Starting wazuh-indexer: /bin/systemctl
/etc/init.d/wazuh-indexer: line 89: daemon: command not found

Service stop wazuh-indexer
Service status wazuh-indexer
/etc/init.d/wazuh-indexer: line 124: status: command not found
```

## Test 2
To test if the removal of the files solves the errors, a new test has been done, including the possible solution
### Test 2 Code

```
WAZUH_MANAGER="10.0.0.2"
for COMPONENT in "wazuh-agent" "wazuh-manager" "wazuh-indexer" "wazuh-dashboard"
do
    yum install -y ./$COMPONENT*
    if [ "$COMPONENT" = "wazuh-indexer" ]
    then
        rm -f /etc/init.d/wazuh-indexer
    elif [ ! "$COMPONENT" = "wazuh-dashboard" ]
    then
        rm -f /etc/rc.d/init.d/$COMPONENT
    fi
    systemctl daemon-reload
    echo "Enable $COMPONENT"
    systemctl enable $COMPONENT
    echo "Disable $COMPONENT"
    systemctl disable $COMPONENT
    echo "Start $COMPONENT"
    systemctl start $COMPONENT
    echo "Restart $COMPONENT"
    systemctl restart $COMPONENT
    echo "Stop $COMPONENT"
    systemctl stop $COMPONENT
    echo "Service start $COMPONENT"
    service $COMPONENT start
    echo "Service stop $COMPONENT"
    service $COMPONENT stop
    echo "Service status $COMPONENT"
    service $COMPONENT status
    yum remove -y $COMPONENT
done
```

### Test 2 results

<details>
  <summary>Wazuh agent</summary>

```
Enable wazuh-agent
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-agent.service → /usr/lib/systemd/system/wazuh-agent.service.
Disable wazuh-agent
Removed "/etc/systemd/system/multi-user.target.wants/wazuh-agent.service".
Start wazuh-agent
Job for wazuh-agent.service failed because the control process exited with error code.
See "systemctl status wazuh-agent.service" and "journalctl -xeu wazuh-agent.service" for details.
Restart wazuh-agent
Job for wazuh-agent.service failed because the control process exited with error code.
See "systemctl status wazuh-agent.service" and "journalctl -xeu wazuh-agent.service" for details.
Stop wazuh-agent
Service start wazuh-agent
Redirecting to /bin/systemctl start wazuh-agent.service
Job for wazuh-agent.service failed because the control process exited with error code.
See "systemctl status wazuh-agent.service" and "journalctl -xeu wazuh-agent.service" for details.
Service stop wazuh-agent
Redirecting to /bin/systemctl stop wazuh-agent.service
Service status wazuh-agent
Redirecting to /bin/systemctl status wazuh-agent.service
× wazuh-agent.service - Wazuh agent
     Loaded: loaded (/usr/lib/systemd/system/wazuh-agent.service; disabled; preset: disabled)
     Active: failed (Result: exit-code) since Fri 2023-02-03 11:39:51 UTC; 119ms ago
    Process: 18539 ExecStart=/usr/bin/env /var/ossec/bin/wazuh-control start (code=exited, status=1/FAILURE)
        CPU: 22ms

Feb 03 11:39:51 centos9stream systemd[1]: Starting Wazuh agent...
Feb 03 11:39:51 centos9stream env[18547]: 2023/02/03 11:39:51 wazuh-agentd: ERROR: (4112): Invalid server address found: 'MANAGER_IP'
Feb 03 11:39:51 centos9stream env[18547]: 2023/02/03 11:39:51 wazuh-agentd: CRITICAL: (1215): No client configured. Exiting.
Feb 03 11:39:51 centos9stream env[18539]: wazuh-agentd: Configuration error. Exiting
Feb 03 11:39:51 centos9stream systemd[1]: wazuh-agent.service: Control process exited, code=exited, status=1/FAILURE
Feb 03 11:39:51 centos9stream systemd[1]: wazuh-agent.service: Failed with result 'exit-code'.
Feb 03 11:39:51 centos9stream systemd[1]: Failed to start Wazuh agent.
```
</details>

<details>
  <summary>Wazuh manager</summary>

```
Enable wazuh-manager
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-manager.service → /usr/lib/systemd/system/wazuh-manager.service.
Disable wazuh-manager
Removed "/etc/systemd/system/multi-user.target.wants/wazuh-manager.service".
Start wazuh-manager
Restart wazuh-manager
Stop wazuh-manager
Service start wazuh-manager
Redirecting to /bin/systemctl start wazuh-manager.service
Service stop wazuh-manager
Redirecting to /bin/systemctl stop wazuh-manager.service
Service status wazuh-manager
Redirecting to /bin/systemctl status wazuh-manager.service
○ wazuh-manager.service - Wazuh manager
     Loaded: loaded (/usr/lib/systemd/system/wazuh-manager.service; disabled; preset: disabled)
     Active: inactive (dead)

Feb 03 11:42:15 centos9stream env[21007]: Killing wazuh-authd...
Feb 03 11:42:16 centos9stream env[21007]: wazuh-agentlessd not running...
Feb 03 11:42:16 centos9stream env[21007]: wazuh-integratord not running...
Feb 03 11:42:16 centos9stream env[21007]: wazuh-dbd not running...
Feb 03 11:42:16 centos9stream env[21007]: wazuh-csyslogd not running...
Feb 03 11:42:16 centos9stream env[21007]: Killing wazuh-apid...
Feb 03 11:42:16 centos9stream env[21007]: Wazuh v4.5.0 Stopped
Feb 03 11:42:16 centos9stream systemd[1]: wazuh-manager.service: Deactivated successfully.
Feb 03 11:42:16 centos9stream systemd[1]: Stopped Wazuh manager.
Feb 03 11:42:16 centos9stream systemd[1]: wazuh-manager.service: Consumed 19.168s CPU time.
```
</details>

<details>
  <summary>Wazuh indexer</summary>

```
Enable wazuh-indexer
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-indexer.service → /usr/lib/systemd/system/wazuh-indexer.service.
Disable wazuh-indexer
Removed "/etc/systemd/system/multi-user.target.wants/wazuh-indexer.service".
Start wazuh-indexer
Job for wazuh-indexer.service failed because the control process exited with error code.
See "systemctl status wazuh-indexer.service" and "journalctl -xeu wazuh-indexer.service" for details.
Restart wazuh-indexer
Job for wazuh-indexer.service failed because the control process exited with error code.
See "systemctl status wazuh-indexer.service" and "journalctl -xeu wazuh-indexer.service" for details.
Stop wazuh-indexer
Service start wazuh-indexer
Redirecting to /bin/systemctl start wazuh-indexer.service
Job for wazuh-indexer.service failed because the control process exited with error code.
See "systemctl status wazuh-indexer.service" and "journalctl -xeu wazuh-indexer.service" for details.
Service stop wazuh-indexer
Redirecting to /bin/systemctl stop wazuh-indexer.service
Service status wazuh-indexer
Redirecting to /bin/systemctl status wazuh-indexer.service
× wazuh-indexer.service - Wazuh-indexer
     Loaded: loaded (/usr/lib/systemd/system/wazuh-indexer.service; disabled; preset: disabled)
     Active: failed (Result: exit-code) since Fri 2023-02-03 11:46:06 UTC; 4s ago
       Docs: https://documentation.wazuh.com
    Process: 21834 ExecStart=/usr/share/wazuh-indexer/bin/systemd-entrypoint -p ${PID_DIR}/wazuh-indexer.pid --quiet (code=exited, status=1/FAILURE)
   Main PID: 21834 (code=exited, status=1/FAILURE)
        CPU: 13.191s
```
</details>

<details>
  <summary>Wazuh dashboard</summary>

```
Enable wazuh-dashboard
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-dashboard.service → /etc/systemd/system/wazuh-dashboard.service.
Disable wazuh-dashboard
Removed "/etc/systemd/system/multi-user.target.wants/wazuh-dashboard.service".
Start wazuh-dashboard
Restart wazuh-dashboard
Stop wazuh-dashboard
Service start wazuh-dashboard
Redirecting to /bin/systemctl start wazuh-dashboard.service
Service stop wazuh-dashboard
Redirecting to /bin/systemctl stop wazuh-dashboard.service
Service status wazuh-dashboard
Redirecting to /bin/systemctl status wazuh-dashboard.service
○ wazuh-dashboard.service - wazuh-dashboard
     Loaded: loaded (/etc/systemd/system/wazuh-dashboard.service; disabled; preset: disabled)
     Active: inactive (dead)

Feb 03 11:48:33 centos9stream systemd[1]: wazuh-dashboard.service: Deactivated successfully.
Feb 03 11:48:33 centos9stream systemd[1]: Stopped wazuh-dashboard.
Feb 03 11:48:33 centos9stream systemd[1]: Started wazuh-dashboard.
Feb 03 11:48:33 centos9stream systemd[1]: Stopping wazuh-dashboard...
Feb 03 11:48:33 centos9stream systemd[1]: wazuh-dashboard.service: Deactivated successfully.
Feb 03 11:48:33 centos9stream systemd[1]: Stopped wazuh-dashboard.
Feb 03 11:48:33 centos9stream systemd[1]: Started wazuh-dashboard.
Feb 03 11:48:33 centos9stream systemd[1]: Stopping wazuh-dashboard...
Feb 03 11:48:33 centos9stream systemd[1]: wazuh-dashboard.service: Deactivated successfully.
Feb 03 11:48:33 centos9stream systemd[1]: Stopped wazuh-dashboard.
```
</details>

### Test 2 summary
As seen in the test results, the removal(or not installation on the first place) of files `/etc/rc.d/init.d/wazuh-agent`, `/etc/rc.d/init.d/wazuh-manager` and `/etc/init.d/wazuh-indexer` removes all errors seen. 

The next thing to discern is what marks an OS as one in which `init.d` is completely deprecated and this files should be removed. The most clear way would be to look for file `/usr/lib/systemd/systemd-sysv-install` for the installation of the Wazuh agent and manager. In the case of the Wazuh indexer, we could look for `/etc/rc.d/init.d/functions` but it is more probable that those directories have been manually repopulates by the user, than file `/usr/lib/systemd/systemd-sysv-install` been created, so we will use the same condition.

A Pull Request has been created, https://github.com/wazuh/wazuh-packages/pull/2070, in which the changes are added.

**Comment by verdx** [comment]
PR https://github.com/wazuh/wazuh-packages/pull/2070 has been tested in all the systems seen in the initial commentary and using all possible actions with both `systemctl` and `service` and no error has been encountered. It isn't necessary to change the documentation, as the steps shown in it now function correctly.

**Comment by verdx** [comment]
# Update

### Previous code
After @DFolchA's commentary and tests on `init.d`-using systems, such as `SLES 11` and `CentOS 6`, it has been proved all changes simply removed the `init.d` service, no matter the system. The macro used, `%define` was replaced during the package building and not the package install and so the service was never installed.

### Current code architecture
Right now, the idea for a solution on this matter is different on the manager, indexer and agent:
- On the manager and indexer, as all supported systems use `systemd`, the `init.d` service has been completely removed. This way, if any new system completely deprecates `init.d`, as `Red Hat Linux 9` and its derivatives have done, no problem will arise.
- The main idea for the agent is to remove the `init.d` service on the post installation of the package if the system uses `systemd`. The main problem right now has come in identifying if `systemd` is being used in a correct way for all possible systems. 

### Identifying if `systemd` is being used
There are a few possible ways to try and see if `systemd` is being used
- `ps --no-headers -p 1 -o comm=` returns "systemd" if it is being used
- According to the [documentation](https://www.freedesktop.org/software/systemd/man/sd_booted.html), the way the own `systemctl` checks for it is by doing this check `[[ -d /run/systemd/system ]]`, whether directory `/run/systemd/system` exists.
- Other option is checking if command `systemctl` is present, but it is not a very clean way, and the command can be removed. 

The main problem is docker, as this methods work on normal hosts, but docker containers do not really use `systemd`, so the real accurate methods don't really work, which is a problem when doing tests. 


## New GitHub Action tests
To test the different options that affect this PR, new tests have been added, mainly adding new systems to the install and enable test and also trying to test the init.d systems too, although it isn't clear this tests can be done or are really useful. They are all tests done on Docker, which, for `init` tests, is not at all the best platform.


(This commentary was accidentally written on the PR instead of the Issue the day before, as seen [here](https://github.com/wazuh/wazuh-packages/pull/2070#issuecomment-1421122559))

**Comment by uiteindelijkwordtallesbagger** [comment]
should Debian 11 and upward be added to this list?

**Comment by verdx** [comment]
# Update
It has been decided the problems with Docker do not affect the correct functioning of the App, and so the check of `[[ -d /run/systemd/system ]]` is going to be the way of checking if `systemd` is being used. It has also been decided that even though all supported systems for the manager use `systemd`, the support for `init.d` is going to be left.

## Testing
The new solution has been tested on all systems in the list at the Issue description. A new system CentOS 6, has been added, so as to test in at least one system using `init.d`. On CentOS 6, the output of the test has been the following. As can be seen, both the Wazuh indexer and dashboard fail on CentOS 6, because none of them install the service for `init.d`. This comes from Opensearch and Kibana, respectively. 

<details>

```
[root@centos6 1918]# bash test.sh
Loaded plugins: fastestmirror
Setting up Install Process
Examining ./wazuh-indexer-4.5.0-1.x86_64.rpm: wazuh-indexer-4.5.0-1.x86_64
Marking ./wazuh-indexer-4.5.0-1.x86_64.rpm to be installed
Loading mirror speeds from cached hostfile
base                                                                   | 3.7 kB     00:00     
epel                                                                   | 4.7 kB     00:00     
extras                                                                 | 3.4 kB     00:00     
updates                                                                | 3.4 kB     00:00     
wazuh                                                                  | 3.4 kB     00:00     
Resolving Dependencies
--> Running transaction check
---> Package wazuh-indexer.x86_64 0:4.5.0-1 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================
 Package              Arch          Version        Repository                            Size
==============================================================================================
Installing:
 wazuh-indexer        x86_64        4.5.0-1        /wazuh-indexer-4.5.0-1.x86_64        747 M

Transaction Summary
==============================================================================================
Install       1 Package(s)

Total size: 747 M
Installed size: 747 M
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Installing : wazuh-indexer-4.5.0-1.x86_64                                               1/1 
/var/tmp/rpm-tmp.X1tCwV: /usr/share/wazuh-indexer/bin/opensearch-keystore: /usr/bin/bash: bad interpreter: No such file or directory
chown: cannot access `/etc/wazuh-indexer/opensearch.keystore': No such file or directory
chmod: cannot access `/etc/wazuh-indexer/opensearch.keystore': No such file or directory
md5sum: /etc/wazuh-indexer/opensearch.keystore: No such file or directory
  Verifying  : wazuh-indexer-4.5.0-1.x86_64                                               1/1 

Installed:
  wazuh-indexer.x86_64 0:4.5.0-1                                                              

Complete!
init
########Service start wazuh-indexer
Starting wazuh-indexer: bash: /usr/share/wazuh-indexer/bin/opensearch: /usr/bin/bash: bad interpreter: No such file or directory

########Service stop wazuh-indexer
########Service status wazuh-indexer
wazuh-indexer dead but pid file exists
Loaded plugins: fastestmirror
Setting up Remove Process
Resolving Dependencies
--> Running transaction check
---> Package wazuh-indexer.x86_64 0:4.5.0-1 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================
 Package             Arch         Version          Repository                            Size
==============================================================================================
Removing:
 wazuh-indexer       x86_64       4.5.0-1          @/wazuh-indexer-4.5.0-1.x86_64       747 M

Transaction Summary
==============================================================================================
Remove        1 Package(s)

Installed size: 747 M
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
Stopping wazuh-indexer service... OK
  Erasing    : wazuh-indexer-4.5.0-1.x86_64                                               1/1 
  Verifying  : wazuh-indexer-4.5.0-1.x86_64                                               1/1 

Removed:
  wazuh-indexer.x86_64 0:4.5.0-1                                                              

Complete!
Loaded plugins: fastestmirror
Setting up Install Process
Examining ./wazuh-manager-4.5.0-1.x86_64.rpm: wazuh-manager-4.5.0-1.x86_64
Marking ./wazuh-manager-4.5.0-1.x86_64.rpm to be installed
Loading mirror speeds from cached hostfile
base                                                                   | 3.7 kB     00:00     
epel                                                                   | 4.7 kB     00:00     
extras                                                                 | 3.4 kB     00:00     
updates                                                                | 3.4 kB     00:00     
wazuh                                                                  | 3.4 kB     00:00     
Resolving Dependencies
--> Running transaction check
---> Package wazuh-manager.x86_64 0:4.5.0-1 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================
 Package              Arch          Version        Repository                            Size
==============================================================================================
Installing:
 wazuh-manager        x86_64        4.5.0-1        /wazuh-manager-4.5.0-1.x86_64        444 M

Transaction Summary
==============================================================================================
Install       1 Package(s)

Total size: 444 M
Installed size: 444 M
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Installing : wazuh-manager-4.5.0-1.x86_64                                               1/1 
  Verifying  : wazuh-manager-4.5.0-1.x86_64                                               1/1 

Installed:
  wazuh-manager.x86_64 0:4.5.0-1                                                              

Complete!
init
########Service start wazuh-manager
Starting Wazuh: 2023/02/08 13:59:16 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
                                                           [  OK  ]
########Service stop wazuh-manager
Stopping Wazuh:                                            [  OK  ]
########Service status wazuh-manager
wazuh-clusterd not running...
wazuh-modulesd not running...
wazuh-monitord not running...
wazuh-logcollector not running...
wazuh-remoted not running...
wazuh-syscheckd not running...
wazuh-analysisd not running...
wazuh-maild not running...
wazuh-execd not running...
wazuh-db not running...
wazuh-authd not running...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
wazuh-apid not running...
Loaded plugins: fastestmirror
Setting up Remove Process
Resolving Dependencies
--> Running transaction check
---> Package wazuh-manager.x86_64 0:4.5.0-1 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================
 Package             Arch         Version          Repository                            Size
==============================================================================================
Removing:
 wazuh-manager       x86_64       4.5.0-1          @/wazuh-manager-4.5.0-1.x86_64       444 M

Transaction Summary
==============================================================================================
Remove        1 Package(s)

Installed size: 444 M
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Erasing    : wazuh-manager-4.5.0-1.x86_64                                               1/1 
warning: /var/ossec/etc/ossec.conf saved as /var/ossec/etc/ossec.conf.rpmsave
  Verifying  : wazuh-manager-4.5.0-1.x86_64                                               1/1 

Removed:
  wazuh-manager.x86_64 0:4.5.0-1                                                              

Complete!
Loaded plugins: fastestmirror
Setting up Install Process
Examining ./wazuh-agent-4.5.0-1.x86_64.rpm: wazuh-agent-4.5.0-1.x86_64
Marking ./wazuh-agent-4.5.0-1.x86_64.rpm to be installed
Loading mirror speeds from cached hostfile
base                                                                   | 3.7 kB     00:00     
epel                                                                   | 4.7 kB     00:00     
extras                                                                 | 3.4 kB     00:00     
updates                                                                | 3.4 kB     00:00     
wazuh                                                                  | 3.4 kB     00:00     
Resolving Dependencies
--> Running transaction check
---> Package wazuh-agent.x86_64 0:4.5.0-1 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================
 Package             Arch           Version         Repository                           Size
==============================================================================================
Installing:
 wazuh-agent         x86_64         4.5.0-1         /wazuh-agent-4.5.0-1.x86_64          25 M

Transaction Summary
==============================================================================================
Install       1 Package(s)

Total size: 25 M
Installed size: 25 M
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Installing : wazuh-agent-4.5.0-1.x86_64                                                 1/1 
  Verifying  : wazuh-agent-4.5.0-1.x86_64                                                 1/1 

Installed:
  wazuh-agent.x86_64 0:4.5.0-1                                                                

Complete!
init
########Service start wazuh-agent
Starting Wazuh:                                            [  OK  ]
########Service stop wazuh-agent
Stopping Wazuh:                                            [  OK  ]
########Service status wazuh-agent
wazuh-modulesd not running...
wazuh-logcollector not running...
wazuh-syscheckd not running...
wazuh-agentd not running...
wazuh-execd not running...
Loaded plugins: fastestmirror
Setting up Remove Process
Resolving Dependencies
--> Running transaction check
---> Package wazuh-agent.x86_64 0:4.5.0-1 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================
 Package            Arch          Version           Repository                           Size
==============================================================================================
Removing:
 wazuh-agent        x86_64        4.5.0-1           @/wazuh-agent-4.5.0-1.x86_64         25 M

Transaction Summary
==============================================================================================
Remove        1 Package(s)

Installed size: 25 M
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Erasing    : wazuh-agent-4.5.0-1.x86_64                                                 1/1 
warning: /var/ossec/etc/ossec.conf saved as /var/ossec/etc/ossec.conf.rpmsave
  Verifying  : wazuh-agent-4.5.0-1.x86_64                                                 1/1 

Removed:
  wazuh-agent.x86_64 0:4.5.0-1                                                                

Complete!
Loaded plugins: fastestmirror
Setting up Install Process
Examining ./wazuh-dashboard-4.5.0-1.x86_64.rpm: wazuh-dashboard-4.5.0-1.x86_64
./wazuh-dashboard-4.5.0-1.x86_64.rpm: does not update installed package.
Error: Nothing to do
init
error reading information on service wazuh-dashboard: No such file or directory
########Service start wazuh-dashboard
wazuh-dashboard: unrecognized service
########Service stop wazuh-dashboard
wazuh-dashboard: unrecognized service
########Service status wazuh-dashboard
wazuh-dashboard: unrecognized service
Loaded plugins: fastestmirror
Setting up Remove Process
Resolving Dependencies
--> Running transaction check
---> Package wazuh-dashboard.x86_64 0:4.5.0-1 will be erased
--> Finished Dependency Resolution

Dependencies Resolved

==============================================================================================
 Package              Arch        Version         Repository                             Size
==============================================================================================
Removing:
 wazuh-dashboard      x86_64      4.5.0-1         @/wazuh-dashboard-4.5.0-1.x86_64      1.1 G

Transaction Summary
==============================================================================================
Remove        1 Package(s)

Installed size: 1.1 G
Downloading Packages:
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
Stopping wazuh-dashboard service...error: %preun(wazuh-dashboard-4.5.0-1.x86_64) scriptlet failed, exit status 1
Error in PREUN scriptlet in rpm package wazuh-dashboard
wazuh-dashboard-4.5.0-1.x86_64 was supposed to be removed but is not!
  Verifying  : wazuh-dashboard-4.5.0-1.x86_64                                             1/1 

Failed:
  wazuh-dashboard.x86_64 0:4.5.0-1                                                            
```
</details>

## GitHub Actions

As the new solution doesn't work when the package is installed inside a Docker, and until a different solution for the tests, all testing of the enabling or anything related to services has been removed from the tests, specifically the `Install and enable` test

**Comment by verdx** [comment]
> should Debian 11 and upward be added to this list?

Hi @uiteindelijkwordtallesbagger. For now, there hasn't been any problem with the `DEB` based distributions related to `init.d` and `systemd`. As you can see, they are keeping the support for `init.d` still:
```
root@debian11:/vagrant# ls -la /usr/lib/systemd/systemd-sysv-install
-rwxr-xr-x 1 root root 1360 Aug  7  2022 /usr/lib/systemd/systemd-sysv-install
root@debian11:/vagrant# 
```

**Comment by verdx** [comment]
# Update
The tests for CentOS 5 and 6 have had some problems with the certificates. Only for CentOS 6 has it been possible to solve it dynamically in the script but only when run locally, it isn't clear what is making the error when called from the GitHub Actions. The run in local is:
```
docker run -v /media/verdx/Overdx/Wazuh/entorno/1918/:/packages -v /media/verdx/Overdx/Wazuh/entorno/wazuh-packages/.github/actions/test-install-components/:/vagrant centos:6.9 /bin/bash /vagrant/install_component.sh wazuh-agent-4.5.0-1.x86_64.rpm agent
Installing Wazuh agent.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
107   970  107   970    0     0    984      0 --:--:-- --:--:-- --:--:--  4801
Loaded plugins: fastestmirror, ovl
Setting up Install Process
Examining /packages/wazuh-agent-4.5.0-1.x86_64.rpm: wazuh-agent-4.5.0-1.x86_64
Marking /packages/wazuh-agent-4.5.0-1.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package wazuh-agent.x86_64 0:4.5.0-1 will be installed
--> Processing Dependency: /sbin/service for package: wazuh-agent-4.5.0-1.x86_64
--> Processing Dependency: /sbin/service for package: wazuh-agent-4.5.0-1.x86_64
--> Running transaction check
---> Package initscripts.x86_64 0:9.03.61-1.el6.centos will be installed
--> Processing Dependency: util-linux-ng >= 2.16 for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Processing Dependency: udev >= 125-1 for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Processing Dependency: sysvinit-tools >= 2.87-6 for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Processing Dependency: plymouth for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Processing Dependency: /sbin/pidof for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Processing Dependency: /sbin/ip for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Processing Dependency: /sbin/blkid for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Processing Dependency: /sbin/arping for package: initscripts-9.03.61-1.el6.centos.x86_64
--> Running transaction check
---> Package iproute.x86_64 0:2.6.32-57.el6 will be installed
--> Processing Dependency: iptables >= 1.4.5 for package: iproute-2.6.32-57.el6.x86_64
--> Processing Dependency: libxtables.so.4()(64bit) for package: iproute-2.6.32-57.el6.x86_64
---> Package iputils.x86_64 0:20071127-24.el6 will be installed
---> Package plymouth.x86_64 0:0.8.3-29.el6.centos will be installed
--> Processing Dependency: system-logos for package: plymouth-0.8.3-29.el6.centos.x86_64
--> Processing Dependency: libdrm_radeon.so.1()(64bit) for package: plymouth-0.8.3-29.el6.centos.x86_64
--> Processing Dependency: libdrm_nouveau.so.1()(64bit) for package: plymouth-0.8.3-29.el6.centos.x86_64
--> Processing Dependency: libdrm_intel.so.1()(64bit) for package: plymouth-0.8.3-29.el6.centos.x86_64
--> Processing Dependency: libdrm.so.2()(64bit) for package: plymouth-0.8.3-29.el6.centos.x86_64
---> Package sysvinit-tools.x86_64 0:2.87-6.dsf.el6 will be installed
---> Package udev.x86_64 0:147-2.74.el6_10 will be installed
--> Processing Dependency: hwdata for package: udev-147-2.74.el6_10.x86_64
---> Package util-linux-ng.x86_64 0:2.17.2-12.28.el6_9.2 will be installed
--> Processing Dependency: libuuid = 2.17.2-12.28.el6_9.2 for package: util-linux-ng-2.17.2-12.28.el6_9.2.x86_64
--> Processing Dependency: libblkid = 2.17.2-12.28.el6_9.2 for package: util-linux-ng-2.17.2-12.28.el6_9.2.x86_64
--> Running transaction check
---> Package hwdata.noarch 0:0.233-20.1.el6 will be installed
---> Package iptables.x86_64 0:1.4.7-19.el6 will be installed
--> Processing Dependency: policycoreutils for package: iptables-1.4.7-19.el6.x86_64
---> Package libblkid.x86_64 0:2.17.2-12.28.el6 will be updated
---> Package libblkid.x86_64 0:2.17.2-12.28.el6_9.2 will be an update
---> Package libdrm.x86_64 0:2.4.65-2.el6 will be installed
--> Processing Dependency: libpciaccess.so.0()(64bit) for package: libdrm-2.4.65-2.el6.x86_64
---> Package libuuid.x86_64 0:2.17.2-12.28.el6 will be updated
---> Package libuuid.x86_64 0:2.17.2-12.28.el6_9.2 will be an update
---> Package redhat-logos.noarch 0:60.0.14-12.el6.centos will be installed
--> Running transaction check
---> Package libpciaccess.x86_64 0:0.13.4-1.el6 will be installed
---> Package policycoreutils.x86_64 0:2.0.83-30.1.el6_8 will be installed
--> Processing Dependency: libdbus-glib-1.so.2()(64bit) for package: policycoreutils-2.0.83-30.1.el6_8.x86_64
--> Running transaction check
---> Package dbus-glib.x86_64 0:0.86-6.el6 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package         Arch   Version               Repository                   Size
================================================================================
Installing:
 wazuh-agent     x86_64 4.5.0-1               /wazuh-agent-4.5.0-1.x86_64  25 M
Installing for dependencies:
 dbus-glib       x86_64 0.86-6.el6            C6.10-base                  170 k
 hwdata          noarch 0.233-20.1.el6        C6.10-base                  1.4 M
 initscripts     x86_64 9.03.61-1.el6.centos  C6.10-base                  949 k
 iproute         x86_64 2.6.32-57.el6         C6.10-base                  386 k
 iptables        x86_64 1.4.7-19.el6          C6.10-base                  255 k
 iputils         x86_64 20071127-24.el6       C6.10-base                  121 k
 libdrm          x86_64 2.4.65-2.el6          C6.10-base                  136 k
 libpciaccess    x86_64 0.13.4-1.el6          C6.10-base                   24 k
 plymouth        x86_64 0.8.3-29.el6.centos   C6.10-base                   89 k
 policycoreutils x86_64 2.0.83-30.1.el6_8     C6.10-base                  663 k
 redhat-logos    noarch 60.0.14-12.el6.centos C6.10-base                   15 M
 sysvinit-tools  x86_64 2.87-6.dsf.el6        C6.10-base                   60 k
 udev            x86_64 147-2.74.el6_10       C6.10-updates               358 k
 util-linux-ng   x86_64 2.17.2-12.28.el6_9.2  C6.10-base                  1.6 M
Updating for dependencies:
 libblkid        x86_64 2.17.2-12.28.el6_9.2  C6.10-base                  119 k
 libuuid         x86_64 2.17.2-12.28.el6_9.2  C6.10-base                   72 k

Transaction Summary
================================================================================
Install      15 Package(s)
Upgrade       2 Package(s)

Total size: 46 M
Total download size: 21 M
Downloading Packages:
--------------------------------------------------------------------------------
Total                                           1.3 MB/s |  21 MB     00:16     
warning: rpmts_HdrFromFdno: Header V3 RSA/SHA1 Signature, key ID c105b9de: NOKEY
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
Importing GPG key 0xC105B9DE:
 Userid : CentOS-6 Key (CentOS 6 Official Signing Key) <centos-6-key@centos.org>
 Package: centos-release-6-9.el6.12.3.x86_64 (@CentOS/6.9)
 From   : /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
Running rpm_check_debug
Running Transaction Test
Transaction Test Succeeded
Running Transaction
  Updating   : libuuid-2.17.2-12.28.el6_9.2.x86_64                         1/19 
  Installing : hwdata-0.233-20.1.el6.noarch                                2/19 
  Installing : sysvinit-tools-2.87-6.dsf.el6.x86_64                        3/19 
  Installing : libpciaccess-0.13.4-1.el6.x86_64                            4/19 
  Updating   : libblkid-2.17.2-12.28.el6_9.2.x86_64                        5/19 
  Installing : dbus-glib-0.86-6.el6.x86_64                                 6/19 
  Installing : redhat-logos-60.0.14-12.el6.centos.noarch                   7/19 
  Installing : policycoreutils-2.0.83-30.1.el6_8.x86_64                    8/19 
  Installing : iptables-1.4.7-19.el6.x86_64                                9/19 
  Installing : iproute-2.6.32-57.el6.x86_64                               10/19 
  Installing : libdrm-2.4.65-2.el6.x86_64                                 11/19 
  Installing : iputils-20071127-24.el6.x86_64                             12/19 
  Installing : plymouth-0.8.3-29.el6.centos.x86_64                        13/19 
  Installing : util-linux-ng-2.17.2-12.28.el6_9.2.x86_64                  14/19 
install-info: No such file or directory for /usr/share/info/ipc.info
  Installing : initscripts-9.03.61-1.el6.centos.x86_64                    15/19 
  Installing : udev-147-2.74.el6_10.x86_64                                16/19 
  Installing : wazuh-agent-4.5.0-1.x86_64                                 17/19 
  Cleanup    : libblkid-2.17.2-12.28.el6.x86_64                           18/19 
  Cleanup    : libuuid-2.17.2-12.28.el6.x86_64                            19/19 
  Verifying  : plymouth-0.8.3-29.el6.centos.x86_64                         1/19 
  Verifying  : libblkid-2.17.2-12.28.el6_9.2.x86_64                        2/19 
  Verifying  : redhat-logos-60.0.14-12.el6.centos.noarch                   3/19 
  Verifying  : sysvinit-tools-2.87-6.dsf.el6.x86_64                        4/19 
  Verifying  : initscripts-9.03.61-1.el6.centos.x86_64                     5/19 
  Verifying  : iptables-1.4.7-19.el6.x86_64                                6/19 
  Verifying  : policycoreutils-2.0.83-30.1.el6_8.x86_64                    7/19 
  Verifying  : wazuh-agent-4.5.0-1.x86_64                                  8/19 
  Verifying  : hwdata-0.233-20.1.el6.noarch                                9/19 
  Verifying  : udev-147-2.74.el6_10.x86_64                                10/19 
  Verifying  : util-linux-ng-2.17.2-12.28.el6_9.2.x86_64                  11/19 
  Verifying  : iputils-20071127-24.el6.x86_64                             12/19 
  Verifying  : dbus-glib-0.86-6.el6.x86_64                                13/19 
  Verifying  : libpciaccess-0.13.4-1.el6.x86_64                           14/19 
  Verifying  : libdrm-2.4.65-2.el6.x86_64                                 15/19 
  Verifying  : iproute-2.6.32-57.el6.x86_64                               16/19 
  Verifying  : libuuid-2.17.2-12.28.el6_9.2.x86_64                        17/19 
  Verifying  : libuuid-2.17.2-12.28.el6.x86_64                            18/19 
  Verifying  : libblkid-2.17.2-12.28.el6.x86_64                           19/19 

Installed:
  wazuh-agent.x86_64 0:4.5.0-1                                                  

Dependency Installed:
  dbus-glib.x86_64 0:0.86-6.el6                                                 
  hwdata.noarch 0:0.233-20.1.el6                                                
  initscripts.x86_64 0:9.03.61-1.el6.centos                                     
  iproute.x86_64 0:2.6.32-57.el6                                                
  iptables.x86_64 0:1.4.7-19.el6                                                
  iputils.x86_64 0:20071127-24.el6                                              
  libdrm.x86_64 0:2.4.65-2.el6                                                  
  libpciaccess.x86_64 0:0.13.4-1.el6                                            
  plymouth.x86_64 0:0.8.3-29.el6.centos                                         
  policycoreutils.x86_64 0:2.0.83-30.1.el6_8                                    
  redhat-logos.noarch 0:60.0.14-12.el6.centos                                   
  sysvinit-tools.x86_64 0:2.87-6.dsf.el6                                        
  udev.x86_64 0:147-2.74.el6_10                                                 
  util-linux-ng.x86_64 0:2.17.2-12.28.el6_9.2                                   

Dependency Updated:
  libblkid.x86_64 0:2.17.2-12.28.el6_9.2  libuuid.x86_64 0:2.17.2-12.28.el6_9.2 

Complete!
```

Some Distributed Jenkins tests have also failed, but it is a known issue, https://github.com/wazuh/wazuh-jenkins/issues/4682

**Comment by verdx** [comment]
# Update
### DEB system changes
When trying to implement the same changes of this issue on DEB systems, several errors related to `systemd` and `sysv` have been found in them. A new issue is being created to fix this errors and try and implement the same changes done in this issue.

### GitHub Actions errors
Some errors have been found in the GitHub Actions that upload and build the Docker Images and a new PR has been made to fix them, which blocks this issue

**Comment by verdx** [comment]
Blocked by https://github.com/wazuh/wazuh-packages/pull/2085/checks

**Comment by DFolchA** [comment]
Closed with https://github.com/wazuh/wazuh-packages/pull/2070
